exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/shared/functions/WindowFunctionsTests.kt

```diff
@@ -0,0 +1,436 @@
+package org.jetbrains.exposed.sql.tests.shared.functions
+
+import org.jetbrains.exposed.sql.*
+import org.jetbrains.exposed.sql.SqlExpressionBuilder.cumeDist
+import org.jetbrains.exposed.sql.SqlExpressionBuilder.denseRank
+import org.jetbrains.exposed.sql.SqlExpressionBuilder.firstValue
+import org.jetbrains.exposed.sql.SqlExpressionBuilder.lag
+import org.jetbrains.exposed.sql.SqlExpressionBuilder.lastValue
+import org.jetbrains.exposed.sql.SqlExpressionBuilder.lead
+import org.jetbrains.exposed.sql.SqlExpressionBuilder.minus
+import org.jetbrains.exposed.sql.SqlExpressionBuilder.nthValue
+import org.jetbrains.exposed.sql.SqlExpressionBuilder.ntile
+import org.jetbrains.exposed.sql.SqlExpressionBuilder.percentRank
+import org.jetbrains.exposed.sql.SqlExpressionBuilder.plus
+import org.jetbrains.exposed.sql.SqlExpressionBuilder.rank
+import org.jetbrains.exposed.sql.SqlExpressionBuilder.rowNumber
+import org.jetbrains.exposed.sql.tests.DatabaseTestsBase
+import org.jetbrains.exposed.sql.tests.TestDB.*
+import org.jetbrains.exposed.sql.tests.TestDB.Companion.allH2TestDB
+import org.jetbrains.exposed.sql.tests.shared.assertEqualLists
+import org.jetbrains.exposed.sql.tests.shared.dml.DMLTestsData
+import org.jetbrains.exposed.sql.tests.shared.dml.withSales
+import org.jetbrains.exposed.sql.vendors.currentDialect
+import org.junit.Test
+import java.math.BigDecimal
+import java.math.RoundingMode
+
+class WindowFunctionsTests : DatabaseTestsBase() {
+
+    private val supportsCountDistinctAsWindowFunction = allH2TestDB + ORACLE
+    private val supportsStatisticsAggregateFunctions = values().toList() - listOf(SQLSERVER, SQLITE)
+    private val supportsNthValueFunction = values().toList() - listOf(SQLSERVER)
+    private val supportsExpressionsInWindowFunctionArguments = values().toList() - listOf(MYSQL)
+    private val supportsExpressionsInWindowFrameClause = values().toList() - listOf(MYSQL, SQLSERVER, MARIADB)
+    private val supportsDefaultValueInLeadLagFunctions = values().toList() - listOf(MARIADB)
+    private val supportsRangeModeWithOffsetFrameBound = values().toList() - listOf(SQLSERVER)
+
+    @Suppress("LongMethod")
+    @Test
+    fun testWindowFunctions() = withSales { testDb, sales ->
+        if (currentDialect.supportsWindowFunctions) {
```

Comment:
Is MySQL (versions earlier than 8.0) the only Exposed database not supporting window functions?

If so, please replace these `currentDialect` checks with the more descriptive conditional from DatabaseTestsBase.kt, `isOldMySql()`. That way it becomes immediately obvious which databases are being included/excluded in each test without having to leave the test suite and/or search the individual vendor files for `supportsWindowFunctions` overrides.

Reply:
It is good point, better to use this recently added helper function. I will refactor these conditions.